src/drivers/stm32/drv_io_timer.c

```diff
@@ -423,12 +452,19 @@ int io_timer_init_timer(unsigned timer)
 			rBDTR(timer) = ATIM_BDTR_MOE;
 		}
 
-		/* If the timer clock source provided as clock_freq is the STM32_APBx_TIMx_CLKIN
-		 * then configure the timer to free-run at 1MHz.
-		 * Otherwize, other frequencies are attainable by adjusting .clock_freq accordingly.
```

Comment:
Please add a note on how specific HW can obtain the an alternate frequency by adjusting .clock_freq so folks do not ifdef this again.